### PR TITLE
Add comments and test for SFU components

### PR DESCRIPTION
**Pull request by ipavlidakis**
### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamVideo/Models/CallParticipant.swift [hunk 1]
```diff
@@ -446,7 +446,29 @@ public struct CallParticipant: Identifiable, Sendable, Equatable {
 
 extension Stream_Video_Sfu_Models_Participant {
 
-    func toCallParticipant(showTrack: Bool = true, pin: PinInfo? = nil) -> CallParticipant {
+    init(_ source: CallParticipant) {
```
Comment (martinmitrevski) [inline Sources/StreamVideo/Models/CallParticipant.swift:449]: why do we need this one? If it's for tests, best to move it in the test target.

File: Sources/StreamVideo/Models/ConnectionQuality.swift [hunk 2]
```diff
@@ -13,7 +13,20 @@ public enum ConnectionQuality: Equatable, Sendable {
 }
 
 extension Stream_Video_Sfu_Models_ConnectionQuality {
-    
+
+    init(_ source: ConnectionQuality) {
```
Comment (martinmitrevski) [inline Sources/StreamVideo/Models/ConnectionQuality.swift:17]: Same here, I doubt we need creation of SFU models in `StreamVideo`.

File: Sources/StreamVideo/Utils/ReflectiveStringConvertible/ReflectiveStringConvertible.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A protocol that extends `CustomStringConvertible` to provide reflective string conversion capabilities.
+///
+/// Types conforming to this protocol can customize their string representation by excluding specific properties
+/// and leveraging Swift's reflection capabilities.
+public protocol ReflectiveStringConvertible: CustomStringConvertible {
```
Comment (martinmitrevski) [inline Sources/StreamVideo/Utils/ReflectiveStringConvertible/ReflectiveStringConvertible.swift:11]: This is a replacement for the multiple description implementations we had?

File: Sources/StreamVideo/WebRTC/v2/SFU/Extensions/SFUSignalService.swift [hunk 4]
```diff
@@ -0,0 +1,16 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+final class SFUSignalService {
```
Comment (martinmitrevski) [inline Sources/StreamVideo/WebRTC/v2/SFU/Extensions/SFUSignalService.swift:8]: This is strange, we have 2 of these `SFUSignalService` it seems?

File: Sources/StreamVideo/WebRTC/v2/SFU/SFUEventAdapter.swift [hunk 5]
```diff
@@ -124,10 +150,12 @@ final class SFUEventAdapter {
                 audioLevel: level.level
             )
         }
-        //        stateAdapter.participantsUpdateSubject.send(participants)
         await stateAdapter.didUpdateParticipants(participants)
     }
 
+    /// Handles a ChangePublishQuality event.
```
Comment (martinmitrevski) [inline Sources/StreamVideo/WebRTC/v2/SFU/SFUEventAdapter.swift:156]: We don't have to document the private methods - unless it contains some specific info.

File: StreamVideoTests/WebRTC/SFU/SFUEventAdapter_Tests.swift [hunk 6]
```diff
@@ -0,0 +1,615 @@
...
+            let mockPublisher = try XCTUnwrap(publisher as? MockRTCPeerConnectionCoordinator)
+            let changePublishQualityCall = try XCTUnwrap(mockPublisher.stubbedFunctionInput[.changePublishQuality]?.first)
+
+            switch changePublishQualityCall {
+            case let .changePublishQuality(activeEncodings):
+                XCTAssertEqual(activeEncodings, .init(["q"]))
+            }
+        }
+
+//        let event = Stream_Video_Sfu_Event_ChangePublishQuality.dummy()
```
Comment (martinmitrevski) [inline StreamVideoTests/WebRTC/SFU/SFUEventAdapter_Tests.swift:136]: saw commented out code in few places, let's clean it up

**Review by martinmitrevski — COMMENTED** [review]
Looks good, but it's strange why the SDK size grew so much, can you check this?

File: Sources/StreamVideo/Utils/ReflectiveStringConvertible/ReflectiveStringConvertible.swift (hunk 3 above)
Comment (ipavlidakis) [reply Sources/StreamVideo/Utils/ReflectiveStringConvertible/ReflectiveStringConvertible.swift:11]: That's right

File: Sources/StreamVideo/WebRTC/v2/SFU/Extensions/SFUSignalService.swift (hunk 4 above)
Comment (ipavlidakis) [reply Sources/StreamVideo/WebRTC/v2/SFU/Extensions/SFUSignalService.swift:8]: No we don't. Why do you think that?

File: Sources/StreamVideo/WebRTC/v2/SFU/SFUEventAdapter.swift (hunk 5 above)
Comment (ipavlidakis) [reply Sources/StreamVideo/WebRTC/v2/SFU/SFUEventAdapter.swift:156]: We definitely don't have to but i'm using those comments to note in some cases what the function does. The rest are just for consistency.

File: Sources/StreamVideo/WebRTC/v2/SFU/SFUSignalService.swift [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+import Combine
+import Foundation
+
+/// A final class that implements the `Stream_Video_Sfu_Signal_SignalServer` protocol and
+/// conforms to the `Sendable` protocol.
+///
+/// This class is marked as `@unchecked Sendable` to indicate that it's the developer's responsibility to
+/// ensure thread safety.
+class SFUSignalService: Stream_Video_Sfu_Signal_SignalServer, @unchecked Sendable {
```
Comment (martinmitrevski) [inline Sources/StreamVideo/WebRTC/v2/SFU/SFUSignalService.swift:13]: @ipavlidakis this one here

**Review by martinmitrevski — APPROVED** [review]

**Pull request merged**